### PR TITLE
Optimize j/l/Class.isAssignableFrom on X86

### PR DESCRIPTION
**Pull request by 0dvictor**
Hook java/lang/Class.isAssignableFrom to existing framework for instanceof/checkcast,
so that certain cases can be further optimized.

Signed-off-by: Victor Ding <dvictor@ca.ibm.com>

### CODE REVIEW

File: runtime/compiler/x/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -4504,9 +4504,25 @@ TR::Register *J9::X86::TreeEvaluator::longBitCount(TR::Node *node, TR::CodeGener
...
+   bool needsNULLCHK = false;
+   switch (node->getOpCodeValue())
+      {
+      case TR::checkcast:
+         break;
+      case TR::checkcastAndNULLCHK:
+         needsNULLCHK = true;
+         break;
+      case TR::instanceof:
+         break;
```
Comment (cathyzhyi) [inline runtime/compiler/x/codegen/J9TreeEvaluator.cpp]: nitpicking. Can we group `TR::instanceof` with `TR::checkcast` since they do the same thing?

Comment (0dvictor) [reply runtime/compiler/x/codegen/J9TreeEvaluator.cpp]: Sure, updated.

**Comment by cathyzhyi** [comment]
LGTM

**Comment by fjeremic** [comment]
Jenkins test sanity xlinux

**Comment by 0dvictor** [comment]
JDK11 failure due to #2536.

**Pull request merged**